src/RepoBackend.ts

```diff
@@ -555,6 +566,14 @@ export class RepoBackend {
           this.open(msg.id);
           break;
         }
+        case "DocumentMessage": {
+          const { id, contents } = msg
```

Comment:
This is a case where `id` confusingly ambiguous. If you think of it as a document id, the following code doesn't make sense. If you think of it as a document id, other parts of the code don't make sense!

Reply:
I could be misreading, but it seems this is looking up the collaborating actors for a given doc id incorrectly. Maybe be something like:
```ts
const docBackend = this.docs.get(id)
if (docBackend) {
  const documentActor = this.actor(docBackend.actorId)
  // ...
}
```

I could imagine the `DocBackend` having a method that returns the collaborating actors to simplify this a bit.

Reply:
Fixed. Yeah, this stuff is kind of messy. I think using explicit actor & document ID types would help differentiate all this (and find places where we've conflated them.)